tests/test_overrides_parser.py

```diff
@@ -84,6 +84,8 @@ def eq(item1: Any, item2: Any) -> bool:
         # bool
         param("true", True, id="value:bool"),
         param(".", ".", id="value:dot"),
+        param("|", "|", id="value:pipe"),
+        param("a|b", "a|b", id="value:a-pipe-b"),
```

Comment:
To be consistent with testing of over special characters in unquoted strings, I'd suggest to just add | to UNQUOTED_SPECIAL (https://github.com/facebookresearch/hydra/blob/f797eadc5d3245a282829ce9e50899ced0c035a4/tests/test_overrides_parser.py#L34) instead of adding these two tests.

Reply:
Done in https://github.com/facebookresearch/hydra/pull/1850/commits/96df2e0c94a5f32827a6c3adc0d9a6a7999f4abd. And I can confirm that the tests fail on the `main` branch.